src/styles/main.scss

```diff
@@ -1,3 +1,70 @@
+$item-size: 100px;
+$block-height: 100px;
+$gap: 1px;
+$start-days: (
+  mon: 0,
+  tue: 1,
+  wed: 2,
+  thu: 3,
+  fri: 4,
+  sat: 5,
+  sun: 6
+);
+
+@each $start-day, $start-day-index in $start-days {
+  .calendar-block--start--day--#{$start-day} .calendar-block__day:first-child {
+    margin-left: ($item-size + $gap) * $start-day-index;
+  }
+}
+
 body {
-  margin: 0;
+  display: flex;
+  margin: 30px;
+  justify-content: center;
+}
+
+.calendar-block {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  box-sizing: border-box;
+  width: ($item-size + $gap) * 7;
+
+  &__day {
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i}) {
+        width: $item-size;
+        height: $item-size;
+        background-color: #eee;
+        border: 1px solid black;
+        box-sizing: border-box;
+        position: relative;
+      }
+
+      &:hover {
+        cursor: pointer;
+        background-color: pink;
+        transition: all 0.5s ease;
+        transform: translateY(-20px);
+      }
+
+      &:nth-child(#{$i})::before {
+        width: 100%;
+        height: 100%;
+        display: flex;
+        font-family: Arial, Helvetica, sans-serif;
+        font-size: 30px;
+        justify-content: center;
+        align-items: center;
+        content: $i + "";
+      }
+    }
```

Comment:
you don't need a loop for all this styles. give them without it, but use loop for content:
```scss
 @for $i from 1 to 31 {
      &:nth-child(#{$i})::before {
        content: '#{$i}';
      }
    }
```

Reply:
the rest styles should be applied for `&__day` class without loop, just:
```scss
&__day {
   ...
}
```